zcash_primitives/src/transaction/builder.rs

```diff
@@ -538,6 +538,8 @@ impl<'a, P: consensus::Parameters, U: sapling::builder::ProverProgress> Builder<
     ///
     /// This fee is a function of the spends and outputs that have been added to the builder,
     /// pursuant to the specified [`FeeRule`].
+    ///
+    /// Any ephemeral inputs or outputs are *not* taken into account.
```

Comment:
I don't think that this is relevant at this level? When you call `get_fee` it's relative to the state of the builder at the time; this builder only contains concrete inputs and outputs, and change must be handled by providing an explicit output, so I think this comment is both incorrect and misleading.